### PR TITLE
42.suite configurable hold

### LINKED ISSUE #42: allow suite hold cycle time to be configured in suite.rc
Currently you can specify a hold cycle time on the command line (or via gcylc) at suite start-up.  We should be able to do the same in the suite.rc file, as for final cycle time.  Holding a suite at a certain point rather than stopping it outright allows users to view final task logs etc. via gcylc.

### PR DESCRIPTION
**Pull request by arjclark**
Closes #42.

@hjoliver - please review.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
(I would make the new setting a list. You may want to have multiple hold points in a suite.)

**Comment by arjclark** [comment]
@matthewrmshin - this is just for replicating the CLI functionality in a suite.rc file. To use a list as you suggest would require an extension of the hold mechanism not currently *straightforwardly* possible without some adjustments to the guts of scheduler.py and its interaction with the task pool. If you like, I can raise an issue for adding that kind of functionality in future?

**Comment by arjclark** [comment]
(which is not to say that I don't like the idea)

**Comment by arjclark** [comment]
Would also need some careful thought as to interactions with external hold invocations as well.

File: lib/cylc/scheduler.py [hunk 1]
```diff
@@ -697,10 +697,18 @@ def configure_suite(self, reconfigure=False):
...
+            if self.config.cfg['scheduling']['hold after point']:
+                self._pool_hold_point = get_point(
+                    self.config.cfg['scheduling']['hold after point'])
+
             if self.options.hold_point_string:
                 self._pool_hold_point = get_point(
                     self.options.hold_point_string)
 
+            if self._pool_hold_point:
+                print "[info] suite will hold after " + str(self._pool_hold_point)
```
Comment (hjoliver) [inline lib/cylc/scheduler.py]: This stdout string prefix is not used in cylc.  I would like rationalize all stdout, stderr (along with logging) and consistently use "[info]", "[warn]", etc. ... but we haven't done that yet.  (Actually, there is one other instance put in recently, for "[info] copying across old suite database...")

**Comment by hjoliver** [comment]
Looks good, works, just one minor point above.

File: lib/cylc/scheduler.py (hunk 1 above)
Comment (arjclark) [inline lib/cylc/scheduler.py]: Yeah, that's me copying the style from the above "[info] copying across...". Removing from both.

**Comment by arjclark** [comment]
@hjoliver - corrected.

**Comment by hjoliver** [comment]
@matthewrmshin - regarding "I would make the new setting a list", I agree with @arjclark.  We can always consider your suggestion as a future enhancement of course ...

**Pull request merged**